lib/iiif/image.rb

```diff
@@ -6,18 +6,22 @@ class Image
     #   @return [String]
     # @!attribute [r] source_format
     #   @return [Symbol]
-    attr_reader :path, :source_format
+    # @!attribute [r] id
+    #   @return [String]
+    attr_reader :path, :source_format, :id
 
     ##
     # @param [String] path  a file path to the local copy of the image
     # @param [Symbol] source_format  a format symbol
+    # @param [String] id  the @id URI of the image (to be used in the info.json)
     # @param [Hash] info  the image info to be used in JSON responses
-    # @param [#extract] extracter  an extracter to create an info hash 
+    # @param [#extract] extractor  an extractor to create an info hash
     #   for this image
-    def initialize(path, source_format, info: nil, 
+    def initialize(path, source_format, id, info: nil,
```

Comment:
id is the URI that goes in `@id` (i.e. `{scheme}://{server}{/prefix}/{identifier}`). This is easiest to get from the env, but I think we should avoid passing the whole env through and calculate it upstream--it may also be useful for caching. Agree?